src/ast_traverse.ml

```diff
@@ -94,8 +94,26 @@ class map_with_expansion_context =
   object (self)
     inherit [Expansion_context.Base.t] map_with_context as super
 
-    method! expression ctxt expr =
-      super#expression (Expansion_context.Base.enter_expr ctxt) expr
+    method! expression ctxt
+        { pexp_desc; pexp_loc; pexp_loc_stack; pexp_attributes } =
+      let ctxt = Expansion_context.Base.enter_expr ctxt in
+      let pexp_desc =
+        match pexp_desc with
+        | Pexp_letmodule (name, module_expr, body) ->
+            let name = self#loc (self#option self#string) ctxt name in
+            let module_expr =
+              self#module_expr
+                (ec_enter_module_opt ~loc:name.loc name.txt ctxt)
```

Comment:
Why are you using `name.loc` here instead of `module_expr.pmod_loc`?

```suggestion
                (ec_enter_module_opt ~loc:module_expr.pmod_loc name.txt ctxt)
```